examples/callcc.hvm

```diff
@@ -0,0 +1,10 @@
+// TODO: Doesn't yet work because of a bug in hvm-lang.
```

Comment:
What is the bug that prevents this program from working?

Reply:
The code `main = let * = λ$x *; $x` triggers a `thread 'main' panicked at 'internal error: entered unreachable code', src/net/net_to_hvmc.rs:77:22` panic. The callcc code is an example of this bug because it erases `$garbage` with a let statement.

Reply:
Should the let body not be pruned out when a channel is defined inside of it?

Reply:
Yes, the let shouldn't be removed in this case.

Reply:
@developedby I figured out a workaround that makes the program work in the current version of HVM-lang. 

By adding a `Seq a b = a` rule, and passing `$garbage` as the second argument and `result` as the first one, it's possible to avoid the `let` body being deleted while at the same time discarding `$garbage`